db/client.go

```diff
@@ -0,0 +1,45 @@
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/jbarzegar/ron-mod-manager/config"
+	"github.com/jbarzegar/ron-mod-manager/ent"
+	"github.com/jbarzegar/ron-mod-manager/ent/archive"
+)
+
+var _Client *ent.Client = nil
+
+func Client() *ent.Client {
+	if _Client == nil {
+		log.Fatalf("Client is not initialized")
+		return nil
+	}
+
+	return _Client
+}
```

Comment:
I'd suggest looking at the [`sync`](https://pkg.go.dev/sync) package. It has a lot of nice concurrency state functions as well as one that would be useful here, [`sync.Once`](https://pkg.go.dev/sync#Once).